Framework/DataHandling/inc/MantidDataHandling/SetSample.h

```diff
@@ -49,13 +50,18 @@ class MANTID_DATAHANDLING_DLL SetSample final : public API::Algorithm {
 
   const Geometry::SampleEnvironment *
   setSampleEnvironment(API::MatrixWorkspace_sptr &workspace,
-                       const Kernel::PropertyManager &args);
+                       const Kernel::PropertyManager_sptr &args);
```

Comment:
I'm wondering why pass shared pointers as references? Also, would it be possible to use `PropertyManager_const_sptr` here? Only some getters are used in the method as far as I can see. Same for the `setSampleShape` method and maybe others in this class.